### ISSUE TITLE
no-invalid-hex tries to validate id-selectors if used with @at-root

### ISSUE THREAD

**Issue by jens-duttke** [sasstools/sass-lint#1255]
**What version of Sass Lint are you using?**
1.12.1

**Please include any relevant parts of your configuration**
```
no-invalid-hex: 2
```

**What did you do? Please include the actual source code causing the issue.**
```
#parent-element {
  color: red;
  
  @at-root #child-element {
    color: green;
  }
}
```

**What did you expect to happen?**
Since it's an ID and no hexadecimal value, I expect that no error is shown.  

The sense of the code is, to indicate in the Sass code that `#child-element` is a sub-element of `#parent-element`, but in the final CSS code, it's not required to specify `#parent-element #child-element { ... }`, since `#child-element` is unique.

**What actually happened? Please include any error messages given to you by Sass Lint.**
```
Hexadecimal values must be a valid format [no-invalid-hex]
```

**If you're using a IDE plugin have you tried the CLI too?**
yes
